### PR TITLE
Adjust App::className() to be clean nullable return.

### PR DESCRIPTION
**Pull request by dereuromark**
And follow up fixes.

### CODE REVIEW

File: src/ORM/Association.php [hunk 1]
```diff
@@ -1103,6 +1103,7 @@ protected function _extractFinder($finderData)
      * @param string $alias The alias name you want to get.
      * @param array $options Table options array.
      * @return string
+     * @throws \Exception If class does not exist.
```
Comment (ADmad) [inline src/ORM/Association.php]: \RuntimeException

File: src/ORM/Association.php [hunk 2]
```diff
@@ -1111,6 +1112,11 @@ protected function _getClassName($alias, array $options = [])
         }
 
         $className = App::className($options['className'], 'Model/Table', 'Table') ?: 'Cake\ORM\Table';
+        if (!$className) {
+            throw new RuntimeException(
+                sprintf('Class for %s could not be found.', $options['className'])
+            );
+        }
```
Comment (markstory) [inline src/ORM/Association.php]: Can this error ever be raised? The `?:` should make it impossible.

Comment (inoas) [reply src/ORM/Association.php]: Can we make automagic tables and/or entities opt-out for 4.x through Configure::?

Comment (dereuromark) [reply src/ORM/Association.php]: Fixed

Comment (markstory) [reply src/ORM/Association.php]: @inoas Possibly. It could be an opt-in mode that makes debugging auto-table issues easier.

**Pull request merged**